src/common/valnum.cpp

```diff
@@ -217,32 +271,41 @@ wxIntegerValidatorBase::FromString(const wxString& s, LongestValueType *value)
 bool
 wxIntegerValidatorBase::IsCharOk(const wxString& val, int pos, wxChar ch) const
 {
-    // We may accept minus sign if we can represent negative numbers at all.
-    if ( ch == '-' )
+    if ( ch < '0' || ch > '9' )
```

Comment:
Not sure what exactly has changed here... the logic seems to be the same as before, but the old version seems more natural to me.

Reply:
Agreed, I just reverted to the old version and indeed it is more natrual and understandable than this.